services/repository/files/content.go

```diff
@@ -101,6 +101,22 @@ func GetContentsOrList(ctx context.Context, repo *repo_model.Repository, treePat
 	return fileList, nil
 }
 
+// GetContentTypeFromTreeEntry check what content is behind it
+func GetContentTypeFromTreeEntry(entry *git.TreeEntry) ContentType {
+	switch {
+	case entry.IsDir():
+		return ContentTypeDir
+	case entry.IsSubModule():
+		return ContentTypeSubmodule
+	case entry.IsExecutable(), entry.IsRegular():
+		return ContentTypeRegular
+	case entry.IsLink():
+		return ContentTypeLink
```

Comment:
Can we also distinguish directory symlinks? See https://try.gitea.io/silverwind/symlink-test for an example.

Reply:
![image](https://user-images.githubusercontent.com/24977596/180272716-70788070-6c3e-4fd6-a67b-dbe0d015bf6d.png)

it's handled as if it's a file link

Reply:
tldr: git has no concept of "directorys" .... so why would they have one for directory-links

gitea web ui does handle it similar: https://try.gitea.io/silverwind/symlink-test/src/branch/master/directory-symlink

Reply:
Yeah, but the UI shows a special icon for this:

<img width="168" alt="image" src="https://user-images.githubusercontent.com/115237/180274369-71ad88b5-e431-4fd3-a41e-5afc76605fbb.png">

It detects whether the destination of the symlink is a file or directory.

Reply:
It might be argued that the requested file is still a symlink, which is true, so I guess I won't block on this.

Reply:
yes we resolve it ... ans so api clients have to do that too -> you should do with the `/contents/` api - and that is already aware of such things